src/WindowsUtils/DebuggerTest.cpp

```diff
@@ -0,0 +1,97 @@
+  MOCK_METHOD(void, OnOutputStringDebugEvent, (const DEBUG_EVENT& event), (override));
+  MOCK_METHOD(void, OnExceptionDebugEvent, (const DEBUG_EVENT& event), (override));
+  MOCK_METHOD(void, OnRipEvent, (const DEBUG_EVENT& event), (override));
+};
+
+TEST(Debugger, LaunchProcess) {
+  constexpr size_t kNumListeners = 10;
+  std::vector<MockListener> mock_listeners(kNumListeners);
+  std::vector<DebugEventListener*> listeners;
+  for (MockListener& mock_listener : mock_listeners) {
+    listeners.push_back(&mock_listener);
+  }
+
+  Debugger debugger(listeners);
+
+  for (MockListener& mock_listener : mock_listeners) {
+    EXPECT_CALL(mock_listener, OnCreateProcessDebugEvent).Times(1);
+    EXPECT_CALL(mock_listener, OnExitProcessDebugEvent).Times(1);
+    EXPECT_CALL(mock_listener, OnBreakpointDebugEvent).Times(1);
+    EXPECT_CALL(mock_listener, OnCreateThreadDebugEvent).Times(testing::AtLeast(1));
+    EXPECT_CALL(mock_listener, OnExitThreadDebugEvent).Times(testing::AtLeast(1));
+    EXPECT_CALL(mock_listener, OnLoadDllDebugEvent).Times(testing::AnyNumber());
+    EXPECT_CALL(mock_listener, OnUnLoadDllDebugEvent).Times(testing::AnyNumber());
+    EXPECT_CALL(mock_listener, OnOutputStringDebugEvent).Times(testing::AnyNumber());
+    EXPECT_CALL(mock_listener, OnExceptionDebugEvent).Times(testing::AnyNumber());
+    EXPECT_CALL(mock_listener, OnRipEvent).Times(testing::AnyNumber());
+  }
+
+  const std::string kArguments = "--sleep_for_ms=20";
+  auto result = debugger.Start(GetTestExecutablePath(), /*working_directory=*/"", kArguments);
+  ASSERT_TRUE(result.has_value());
+
+  Debugger::StartInfo& start_info = result.value();
+  std::string expected_command_line = GetTestExecutablePath().string();
+  if (!kArguments.empty()) expected_command_line += absl::StrFormat(" %s", kArguments);
+  EXPECT_STREQ(start_info.command_line.c_str(), expected_command_line.c_str());
+
+  // Wait for debugger to automatically detach upon process termination.
+  debugger.Wait();
+}
+
+TEST(Debugger, NoListener) {
+  Debugger debugger({});
+  const std::string kArguments = "--sleep_for_ms=20";
+  auto result = debugger.Start(GetTestExecutablePath(), /*working_directory=*/"", kArguments);
+  EXPECT_THAT(result, orbit_test_utils::HasError("Debugger has no debug event listener"));
+}
+
+TEST(Debugger, NonExistingExecutable) {
+  MockListener listener;
+  Debugger debugger({&listener});
+  const std::string executable = R"(C:\non_existing_executable.exe)";
+  auto result = debugger.Start(executable, "", "");
+  EXPECT_THAT(result, orbit_test_utils::HasError("Executable does not exist"));
+}
+
+TEST(Debugger, NonExistingWorkingDirectory) {
+  MockListener listener;
+  Debugger debugger({&listener});
+  const std::string non_existing_working_directory = R"(C:\non_existing_directory)";
+  auto result = debugger.Start(GetTestExecutablePath(), non_existing_working_directory, "");
+  EXPECT_THAT(result, orbit_test_utils::HasError("Working directory does not exist"));
+}
```

Comment:
I think we also need a test for the Detach method. But to be honest, I don't really know how that could look like. So feel free to ignore.

Reply:
You're right, I added it, thanks.